src/engine/ResultTable.h

```diff
@@ -21,6 +25,59 @@ using std::mutex;
 using std::unique_lock;
 using std::vector;
 
+// The local vocabulary for a particular result table. It maps the IDs that are
+// not part of the normal vocabulary
+//
+//
+// It contains a map from
+// (local vocab) ids
+class LocalVocab {
+ public:
+  // Create a new, empty local vocabulary.
+  LocalVocab() {}
+
+  // Prevent accidental copying of a local vocabulary.
+  // TODO: Needed in SparqlExpressionTestHelpers.h:91.
+  // LocalVocab(const LocalVocab&) = delete;
+
+  // Get ID of a word in the local vocabulary. If the word was already
+  // contained, return the already existing ID. If the word was not yet
+  // contained, add it, and return the new ID.
+  [[maybe_unused]] Id getIdAndAddIfNotContained(const std::string& word);
```

Comment:
This [[maybe_unused]] works on the function, not on its return value.
The proper way to do that, is make all the other non-void functions `[[nodiscard]]` (in hindsight this should be the default in C++), and state here why that is missing.

Reply:
Should this return a `LocalVocabIndex`?